src/mbgl/util/chrono.cpp

```diff
@@ -22,8 +22,13 @@ std::string rfc1123(Timestamp timestamp) {
     std::time_t time = std::chrono::system_clock::to_time_t(timestamp);
     std::tm info;
     _gmtime(&time, &info);
-    char buffer[30];
-    snprintf(buffer, 30, "%s, %02d %s %4d %02d:%02d:%02d GMT", week[info.tm_wday], info.tm_mday,
+
+    // Buffer size 30 is OK assuming the year has 4 digits. However, In theory, it might have
+    // more digits. Under gcc 8.3.0 with -Os optimization flag, there is compiler warning
+    // complaining about the buffer size might be too small. Inceasing the buffer to 32 fixes
+    // the warning.
+    char buffer[32];
+    snprintf(buffer, 32, "%s, %02d %s %4d %02d:%02d:%02d GMT", week[info.tm_wday], info.tm_mday,
              months[info.tm_mon], 1900 + info.tm_year, info.tm_hour, info.tm_min, info.tm_sec);
     return buffer;
```

Comment:
This is using `string (const char* s);` which `Copies the null-terminated character sequence (C-string) pointed by s`. Given that  two last symbols contain garbage it's not advisable, so we should either zero the buffer or use `std::string(buffer, 30)`

Reply:
Suppose the year has 4 digits. The 30th char would be '\0'. So 32 guarantees it will work up to 6-digit of year, which should be sufficient.